smoketests/dev-server.sh

```diff
@@ -0,0 +1,65 @@
+#!/usr/bin/env bash
+
+# usage sh ./smoketests/dev-server.sh <n_runs>
+
+
+test_folders=(
+    dev-server
+)
+iterations=100
+
+if [ "$1" != "" ]; then
+    iterations=$1
+fi
+
+function setup() {
+cat << EOL >> ./dev-server/index.js
+console.log('index');
+EOL
+
+cat << EOL >> ./dev-server/webpack.config.js
+var HtmlWebpackPlugin = require('html-webpack-plugin');
+var path = require('path');
+
+module.exports = {
+  entry: './index.js',
+  output: {
+    path: path.resolve(__dirname, './dist'),
+    filename: 'index_bundle.js'
+  },
+  plugins: [new HtmlWebpackPlugin()],
+  devServer: {
+      port: 3000
+  }
+};
+EOL
+
+}
+
+function del_files() {
+    rm -rf ./dev-server/index.js
+    rm -rf ./dev-server/webpack.config.js
+}
+function teardown() {
+    del_files
+    cd ../
+    exit
+}
+
+
+setup
+
+for i in "${test_folders[@]}"; do 
+    echo "============================ RUNNING FOLDER: $i ============================"
+    for j in `seq 1 $iterations`; do
+        echo "============================ ITERATION: $j/$iterations  ====================================="
+        ls ./dev-server/*.smoketest.js | xargs -I{} echo "Running:" {} | tee /dev/tty | cut -d':' -f 2 | xargs -I{} node {}
+        sudo lsof -i :3000 | awk '{print $2}' | grep -v "PID" | xargs -I{} kill {} -9
+        if [ "$?" != "0" ]; then
+            teardown
+        fi
+    done
+    echo "============================ DONE RUNNING $i $iterations times ============================"
+done
+
+teardown
```

Comment:
Please avoid using bash, it is unnecessary, you can rewrite it using JavaScript

Reply:
No

Reply:
Why?

Reply:
/cc @webpack/cli-team Somebody love `bash`? I think using `JavaScript` everywhere is good idea, windows always have bugs with `bash`, we are potentially creating flaky tests, before we have same tests for webpack-dev-server and it was big pain

Reply:
I think bash is fine.. The setup and teardown is written in js, that's good enough, don't you think? scripting with bash is made for these tasks

Reply:
What is the need for a bash if we have javascript, what is benefits? All I see is reading problems, especially for new developers.

Reply:
Let's use bash now and change to js when we are done with smoketests, yes?

Reply:
Why do not rewrite it to JS now before it grows into a big rubbish heap?

Reply:
Concurrency is pretty hard with js, could you have a look @evilebottnawi ? I think it's better to handle spawning and cleaning up with bash, gave js a chance but with bash it's easier to control the processes

Reply:
@evenstensberg What's easier? You can describe the problem?

Reply:
Every spawn is async, so each script will be done at a different time, in addition these operations in the scripts won't be in order, scripting is easier cause they will enforce concurrency by default

Reply:
you can use async/await, or you can rewrite then on jest and using `--runInBand`